libs/common/src/database/model/database.model.ts

```diff
@@ -0,0 +1,44 @@
+import { Transform, Type } from 'class-transformer';
+import { IsInt, IsNotEmpty, IsOptional, IsString, Min } from 'class-validator';
+import { FindOptionsWhere, ObjectID } from 'typeorm';
+import { transformSortType } from './transform.model';
+
+export class FindOptions {
+    @IsInt()
+    @Min(1)
+    @Type(() => Number)
+    page: number;
+
+    @IsInt()
+    @Min(1)
+    @Type(() => Number)
+    max: number;
+
+    @IsOptional()
+    @IsString()
+    @IsNotEmpty()
```

Comment:
Me parece que houve aqui um equivoco e você pretende usar `IsOptional` uma vez que o atributo `search` é opcional. Então eu sugiro remover o `IsNotEmpty`.

```suggestion
    @IsOptional()
    @IsString()
```